### PR TITLE
Extracting common code from user_availability concerns

### PR DESCRIPTION
**Pull request by puradawid**
Removing common code from project/user_availability.rb and membership/user_availability.rb to shared/availability.rb file module.

### CODE REVIEW

File: app/controllers/memberships_controller.rb [hunk 1]
```diff
@@ -7,10 +7,15 @@ class MembershipsController < ApplicationController
...
   expose_decorated(:projects) { projects_repository.all_by_name }
-  expose_decorated(:roles) { roles_repository.by_name }
-  expose_decorated(:users) { current_user.admin? ? users_repository.all_by_name : [current_user] }
+  expose_decorated(:roles) { roles_repository.all_by_name }
+  expose_decorated(:users) do
+    current_user.admin? ? users_repository.all_by_name : [current_user]
+  end
+
+  before_filter :authenticate_admin!,
+                only: [:index, :update, :destroy, :create, :edit],
```
Comment (netguru-hound) [inline app/controllers/memberships_controller.rb]: Align the parameters of a method call if they span more than one line.

File: app/controllers/memberships_controller.rb [hunk 2]
```diff
@@ -42,13 +47,14 @@ def destroy
   protected
 
   def membership_params
-    params.require(:membership)
-      .permit(:starts_at, :ends_at, :project_id, :user_id, :role_id, :billable, :booked)
+    params.require(:membership).
+      permit(:starts_at, :ends_at, :project_id,
+             :user_id, :role_id, :billable, :booked)
```
Comment (netguru-hound) [inline app/controllers/memberships_controller.rb]: Align the parameters of a method call if they span more than one line.

File: app/controllers/memberships_controller.rb [hunk 3]
```diff
@@ -42,13 +47,14 @@ def destroy
   protected
 
   def membership_params
-    params.require(:membership)
-      .permit(:starts_at, :ends_at, :project_id, :user_id, :role_id, :billable, :booked)
+    params.require(:membership).
```
Comment (netguru-hound) [inline app/controllers/memberships_controller.rb]: Place the . on the next line, together with the method name.

File: app/models/settings.rb [hunk 4]
```diff
@@ -9,12 +9,31 @@ class Settings
   class << self
     delegate :inspect, to: :instance
 
+    def initialize
+      @instance = first_or_create
+    end
+
     def method_missing(method, *args)
-      instance.public_send(method, *args)
+      instance.public_send(method, *args) if not locked?
```
Comment (netguru-hound) [inline app/models/settings.rb]: Favor `unless` over `if` for negative conditions.<br>Use `!` instead of `not`.

File: app/models/settings.rb [hunk 5]
```diff
@@ -9,12 +9,31 @@ class Settings
...
+    private
+
+    def lock
+      @locked_instance = true
+      result = yield
+      @locked_instance = false
+      result
+    end
+
+    def locked?
```
Comment (netguru-hound) [inline app/models/settings.rb]: Use `attr_reader` to define trivial reader methods.

File: app/controllers/memberships_controller.rb [hunk 6]
```diff
@@ -42,13 +47,13 @@ def destroy
   protected
 
   def membership_params
-    params.require(:membership)
-      .permit(:starts_at, :ends_at, :project_id, :user_id, :role_id, :billable, :booked)
+    params.require(:membership).permit(:starts_at, :ends_at, :project_id,
+                                       :user_id, :role_id, :billable, :booked)
```
Comment (netguru-hound) [inline app/controllers/memberships_controller.rb]: Align the parameters of a method call if they span more than one line.

**Comment by nekath** [comment]
Sorry for dealy and thanks for your contribution :) 
Unfortunately right now we refactor people application a lot, so this code is no longer valid.
Closing.

**Pull request closed without merging**